components/dashboard/src/data/configurations/configuration-queries.ts

```diff
@@ -108,6 +108,8 @@ export const useConfiguration = (configurationId: string) => {
                 }
                 return true;
             },
+            cacheTime: 1000 * 60 * 10, // 10m
+            staleTime: 1000 * 60 * 10, // 10m
```

Comment:
Any particular reason for these two? Maybe it's helpful to have the `staleTime` a bit lower in case someone else from an org changes a name of a repo or something similar. We can also cache for longer, like an hour IMO.

Reply:
@filiptronicek  Every time user change selected repository, it will trigger a request, so I add this cacheTime and staleTime field, but not that understand what they mean 🙈. Do you think it's unnecessary or we can make it lower to 1 minutes?

Reply:
I also found this quite confusing and have to look it up every now and then 😄.

`staleTime` is essentially how long we consider the data as fine to just use without checking again, like a true cache. After that expires (let's say 5 seconds, which is our [global default](https://github.com/gitpod-io/gitpod/blob/f5127b661152e8040d0806a757862148e0c027c8/components/dashboard/src/data/setup.tsx#L53)), the query tries to use its previous data, but only if `cacheTime` has not passed already. When the cache from cacheTime is used, we also send a request to the server to check if our cached data is still fresh and if it updates, we hot swap it to display the current state.

Therefore, we can be quite a bit more generous with our `cacheTime`s and keep our `staleTime`s shorter, because it will mean that even though some requests will be made, it will be most of the time in the background and the UI will have data, which is most likely also up-to-date.

Reply:
Thank you for explaining 🚀